### PR TITLE
Features/retweet

### PR DESCRIPTION
**Pull request by Draqe**
@nzdrml

### CODE REVIEW

File: app/views/tweets/_form3.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<%= form_for @reply do |form| %>
+
+  <p>
+   <%= form.text_area :text %>
+   <%= form.hidden_field :tweet_id %>
+   <%= form.hidden_field :type %>
```
Comment (nzdrml) [inline app/views/tweets/_form3.html.erb]: remove this, and instead set the type in the `create` method. if there's a `tweet_id`, set the type to `reply`

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

File: app/controllers/tweets_controller.rb [hunk 2]
```diff
@@ -1,39 +1,91 @@
...
+      end
+    end
   end
 
   def destroy
     @user = User.find(params[:user_id])
     @tweet = @user.tweets.find(params[:id])
-    @tweet.destroy
-    redirect_to user_path(@user)
+    if @tweet.destroy
```
Comment (nzdrml) [inline app/controllers/tweets_controller.rb]: whats going on in this if-else block? you're doing the exact same things whether it's destroyed or not

File: app/models/user.rb [hunk 3]
```diff
@@ -1,8 +1,18 @@
 class User < ApplicationRecord
 
   has_many :tweets, dependent: :destroy
+  has_many :replies, class_name: 'Reply'
   has_secure_password
+
+  password_format = /\A
```
Comment (nzdrml) [inline app/models/user.rb]: use allcaps for constants. does the value not fit in one line?

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -6,4 +6,9 @@ class ApplicationController < ActionController::Base
   def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
+
+  def authorize
+    redirect_to root_path,
```
Comment (nzdrml) [inline app/controllers/application_controller.rb]: enclose the arguments of this method in parentheses if you're going to wrap to the next line. consult the style guide for the format.

**Comment by Draqe** [comment]
@nzdrml

File: app/controllers/tweets_controller.rb [hunk 5]
```diff
@@ -55,18 +55,11 @@ def update
   def destroy
     @user = User.find(params[:user_id])
     @tweet = @user.tweets.find(params[:id])
-    if @tweet.destroy
-      if @tweet.type == 'Reply'
-        redirect_to new_tweet_reply_path(@tweet.tweet_id)
-      else
-        redirect_to user_path(current_user)
-      end
+    @tweet.destroy
```
Comment (nzdrml) [inline app/controllers/tweets_controller.rb]: you're still supposed to handle if the destroy event fails

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -8,7 +8,8 @@ def current_user
   end
 
   def authorize
-    redirect_to root_path,
-      alert: 'You must be logged in to access this page' if current_user.nil?
+    redirect_to(
+      root_path,alert: 'You must be logged in to access this page' if current_user.nil?
```
Comment (nzdrml) [inline app/controllers/application_controller.rb]: this is very lazy, conventionally there's supposed to be a space after the comma.

Also did you even test this? are you sure this works?

**Comment by Draqe** [comment]
@nzdrml is this how you do it?

**Comment by Draqe** [comment]
@nzdrml is this how you do it?

File: app/controllers/tweets_controller.rb [hunk 7]
```diff
@@ -1,39 +1,89 @@
...
   end
 
   def destroy
     @user = User.find(params[:user_id])
-    @tweet = @user.tweets.find(params[:id])
-    @tweet.destroy
-    redirect_to user_path(@user)
+    begin
+      @tweet = @user.tweets.find(params[:id])
+      @tweet.destroy
```
Comment (nzdrml) [inline app/controllers/tweets_controller.rb]: solution is over complicated. you were on the right track when you had an if statement here.

```ruby
if @tweet.destroy
  # do something
else
  # do something else, and perhaps display an error
end

File: app/controllers/tweets_controller.rb [hunk 8]
```diff
@@ -1,39 +1,89 @@
...
-    @user = User.find(params[:user_id])
-    @tweet = @user.tweets.create(tweet_params)
-    redirect_to user_path(@user)
+    @tweet = Tweet.new(tweet_params)
+    @tweet.user_id = current_user.id
+    if @tweet.tweet_id.present?
+      @tweet.type = 'Reply'
+    end
+    if @tweet.save
+      if @tweet.type == 'Reply'
```
Comment (nzdrml) [inline app/controllers/tweets_controller.rb]: this is exactly the same as your initial mistake with the destroy method. you're supposed to do something that informs the user that something went wrong when attempting to save the record.

If you looked at a few basic CRUD tutorials you'll see how they do it or get some ideas.

Make sure this is addressed everywhere else in your app as well.

File: db/migrate/20190711072431_drop_tweets_table.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class DropTweetsTable < ActiveRecord::Migration[5.2]
+  def up
+    drop_table :tweets
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
```
Comment (nzdrml) [inline db/migrate/20190711072431_drop_tweets_table.rb]: is this necessary? what happens if it's just `def change` and you try to rollback?

File: test/fixtures/retweets.yml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (nzdrml) [inline test/fixtures/retweets.yml]: delete this file

File: test/models/retweet_test.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (nzdrml) [inline test/models/retweet_test.rb]: delete this file

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

**Comment by Draqe** [comment]
@nzdrml

File: app/controllers/tweets_controller.rb (hunk 8 above)
Comment (nzdrml) [reply app/controllers/tweets_controller.rb]: @Draqe ?

**Pull request merged**